src/main/js/controls/Gantt/Gantt.js

```diff
@@ -287,20 +290,22 @@ class Gantt extends Construct {
      * The content serves as a 1to1 relationship. For rendering
      * multiple data sets respective number of content needs to be provided.
      *
-     * @param {object} content - Gantt content to be removed
+     * @param {object|Array} content - Gantt content to be removed
      * @returns {Gantt} - Gantt instance
      */
     unloadContent(content) {
-        const index = this.tracks.indexOf(content.key);
-        if (index < 0) {
-            throw new Error(errors.THROW_MSG_INVALID_OBJECT_PROVIDED);
-        }
-        this.trackConfig[index].unload(this);
+        contentLoadHandler(content, (i) => {
+            const index = this.tracks.indexOf(i.key);
+            if (index < 0) {
+                throw new Error(errors.THROW_MSG_INVALID_OBJECT_PROVIDED);
+            }
+            this.trackConfig[index].unload(this);
+            this.tracks.splice(index, 1);
+            this.trackConfig.splice(index, 1);
+        });
```

Comment:
Is there a same performance gain during unload as well? I am trying to figure what is the `n` value here and if we need to do this for other types of graph as well.

Reply:
Yes there is similar gains for unload as well. I think it is worth logging an issue to look at all other graphs.